content/6_Advanced/Extend_Euclid.mdx

```diff
@@ -26,15 +26,35 @@ The original Euclidean Algorithm computes $\gcd(a,b)$ and looks like this:
 
 ```cpp
 ll euclid(ll a, ll b) {
-	for (; b; swap(a, b)) {
+	while (b > 0) {
 		ll k = a / b;
 		a -= k * b;
+		swap(a, b)
```

Comment:
this doesn't compile (also, what was wrong with the original version?)

Reply:
wanted to keep it consistent with java and python